### DISCUSSION TITLE
Change output of file export & broken avatar

### DISCUSSION THREAD

**Discussion by crafter23456** [WeblateOrg/weblate#8703] [Q&A]
I have plenty of projects in a local repository (no git etc.) and want them in a specific order exported.
For example that advent project.
File mask: `SpieleOase/lobby/advent/*.properties`
But when I export its in the archive under `project/advent/project/lobby/advent/*`
How can I get rid of that `project/advent/` before the actually needed folders?

Secondly my avatar is broken but I dont know how to fix it 
![image](https://user-images.githubusercontent.com/64259198/215562873-51bd9650-6444-4069-a16d-6173c6d91126.png)

**Comment by nijel** [comment]
The layout of ZIP exports is not customizable - it is always PROJECT/COMPONENT/FULL/PATH/TO.FILE.

The avatar is using gravatar.com or libravatar.org depending on the configuration.

**Comment by crafter23456** [comment]
Is there a smart way to have all components in a file structure which arent secluded by the component directory `PROJECT/FULL/PATH/TO.FILE`? It worked when I had a github repo and uploaded there my files in the correct stucture my project needs.

I have that in my settings: 
AVATAR_URL_PREFIX = 'https://www.gravatar.com/'
ENABLE_AVATARS = True
![image](https://user-images.githubusercontent.com/64259198/215565736-7a39c419-1c63-46ce-875c-102d1cc5272c.png)

**Reply by nijel** [reply]
What is the actual URL of the image?

**Reply by crafter23456** [reply]
/weblate/avatar/24/admin.png

**Reply by nijel** [reply]
Ah, sorry, I forgot that avatars are cached server-side. Any error from downloading them will be visible in the logs.

**Reply by crafter23456** [reply]
`[error] 1382274#1382274: *25636 open() "/var/www/oase/weblate/avatar/24/admin.png" failed (2: No such file or directory)`
My URL_PREFIX is /weblate but the project is located in /weblate-env. There isnt even in that directory something like that.

**Reply by nijel** [reply]
What is your setting for `DATA_DIR` and `CACHE_DIR`?

**Reply by crafter23456** [reply]
DATA_DIR = '/var/www/oase/weblate-env/data'
CACHE_DIR = f"{DATA_DIR}/cache"
![image](https://user-images.githubusercontent.com/64259198/222454469-19f86bf6-020d-44f3-9fe8-3bd94c601909.png)

**Reply by nijel** [reply]
Then there is something wrong - the failed open is using `/var/www/oase/weblate/` as CACHE_DIR.

**Reply by crafter23456** [reply]
Yeah, cause the weblate logs are all in the /root folder too. So something is wrong

**Reply by nijel** [reply]
From which log is that error? It doesn't look like a Python or Weblate error message.

**Reply by crafter23456** [reply]
![image](https://user-images.githubusercontent.com/64259198/222730214-d8deebf5-8b45-4bda-846f-d12c42366f35.png)
![image](https://user-images.githubusercontent.com/64259198/222730253-3f20a516-5e43-478e-9094-cf934f1ab589.png)
every log is there somehow. But the avatar thing is more important. All avatar-cache folder are empty and it locates the wrong one

**Comment by nijel** [comment]
Did you restart both celery and wsgi processes after changing configuration? Anyway the celery log is not relevant for what happens in the webserver, check webserver logs instead.

Also, `CACHES` setting might be related to this (it actually defines where avatars are being cached).

**Reply by crafter23456** [reply]
```
location /weblate/ {
        include uwsgi_params;
        uwsgi_pass unix:/run/uwsgi/djwsgi.sock;
  	    proxy_pass http://138.201.251.203:8000;
}
```
`"LOCATION": os.path.join(DATA_DIR, "avatar-cache"),`

I restarted celery with "celery multi restart" and uwsgi with "service uwsgi restart" but no change
